flashinfer_trace/tests/references/test_top_k_top_p_sampling_from_probs_v163840.py

```
@@ -0,0 +1,92 @@
+def load_definition(name: str) -> Definition:
+    for op_dir in DEFINITIONS_DIR.iterdir():
+        if op_dir.is_dir():
+            def_file = op_dir / f"{name}.json"
+            if def_file.exists():
+                return load_json_file(Definition, def_file)
+    raise FileNotFoundError(f"Definition {name} not found in {DEFINITIONS_DIR}")
+
+
+def compile_reference(reference_code: str):
+    namespace = {"torch": torch, "math": math}
+    exec(reference_code, namespace)
```

Ruff (0.15.10)

[error] Use of `exec` detected

(S102)